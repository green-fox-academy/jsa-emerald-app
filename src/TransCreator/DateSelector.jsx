import React from 'react';
import { MaterialIcons } from '@expo/vector-icons';
import {
  Text,
  DatePicker,
} from 'native-base';
import { View } from 'react-native';
import moment from 'moment';
import PropTypes from 'prop-types';
import styles from './Style';


const DateSelector = ({ transDate, setTransDate }) => (
  <>
    <MaterialIcons name="date-range" style={styles.dateItemIcon} />
    <Text>Date: </Text>
    <View style={styles.dateView}>
      <DatePicker
        defaultDate={new Date()}
        minimumDate={new Date(1990, 1, 1)}
        maximumDate={new Date(2050, 12, 31)}
        locale="en"
        modalTransparent={false}
        animationType="fade"
        androidMode="default"
        placeHolderText={transDate}
        textStyle={{ color: '#5C6BC0' }}
        placeHolderTextStyle={{ color: '#5C6BC0' }}
        onDateChange={(newDate) => setTransDate(moment(newDate).format('DD/MM/YYYY'))}
        disabled={false}
      />
    </View>
  </>
);

DateSelector.propTypes = {
  transDate: PropTypes.string,
  setTransDate: PropTypes.func,
};

DateSelector.defaultProps = {
  transDate: null,
  setTransDate: moment().format('DD/MM/YYYY'),
};

export default DateSelector;
