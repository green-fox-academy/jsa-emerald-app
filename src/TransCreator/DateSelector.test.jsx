import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import DateSelector from './DateSelector';

describe('<DateSelector/>', () => {
  it('render', () => {
    const tree = renderer.create(<DateSelector
      transDate={1581379200}
      setTransDate={jest.fn}
    />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Display Click event', () => {
    const wrapper = shallow(<DateSelector
      transDate={1581379200}
      setTransDate={jest.fn}
    />);
    wrapper.find('#text-datePick-display').simulate('press');
    wrapper.find('#dateTimePicker-dateSelector').simulate('cancel');
    wrapper.find('#dateTimePicker-dateSelector').simulate('confirm');
  });
});
