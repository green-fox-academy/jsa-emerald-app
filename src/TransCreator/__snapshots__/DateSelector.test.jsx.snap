// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateSelector/> render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "#f0f0f0",
      "borderTopWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "height": 35,
      "width": 750,
    }
  }
>
  <Text />
  <View
    style={
      Object {
        "borderColor": "#5C6BC0",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginLeft": 5,
        "paddingBottom": 2,
        "paddingLeft": 7,
        "paddingRight": 7,
        "paddingTop": 2,
      }
    }
  >
    <Text
      id="text-datePick-display"
      onPress={[Function]}
      style={
        Object {
          "color": "grey",
        }
      }
    >
      02/11/2020
    </Text>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      onRequestClose={[Function]}
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
      />
      <View
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        onModalWillHide={[Function]}
        onModalWillShow={[Function]}
        pointerEvents="box-none"
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "margin": 10,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 13,
                "marginBottom": 8,
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "white",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#d5d5d5",
                "borderBottomWidth": 0.5,
                "padding": 14,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#8f8f8f",
                    "fontSize": 13,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#5C6BC0",
                  },
                ]
              }
            >
              Pick a date
            </Text>
          </View>
          <View
            onStartShouldSetResponderCapture={[Function]}
          >
            <View>
              <RCTDatePicker
                date={1581379200000}
                minuteInterval={1}
                mode="date"
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "height": 216,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderColor": "#d5d5d5",
                "borderTopWidth": 0.5,
                "height": 57,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#007ff9",
                    "fontSize": 20,
                    "fontWeight": "normal",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Confirm
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 13,
                "height": 57,
                "justifyContent": "center",
                "marginBottom": 0,
              },
              Object {
                "backgroundColor": "white",
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#007ff9",
                  "fontSize": 20,
                  "fontWeight": "600",
                  "padding": 10,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Cancel
          </Text>
        </View>
      </View>
    </Modal>
  </View>
</View>
`;
